machine/machine_core/machine_virtual.py

```diff
@@ -142,15 +147,21 @@ def stdchannel_redirected(stdchannel, dest_filename):
 
 
 class VirtNetwork:
-    def __init__(self, network=None, image="generic"):
-        self.locked = []
+    hostnet: int
+    image: str
+    last: int
+    locked: List[int] = []
+    network: int
+    offset: int
+
+    def __init__(self, id: Optional[int] = None, image: str = "generic"):
```

Comment:
Your introduction of the `id` parameter here is going to clash with ruff `A` rules.